### PR TITLE
Fix active? and trace_context to properly handle inactive traces

### PR DESCRIPTION
**Pull request by samuel-williams-shopify**
- Fix `Traces.active?` to return false when there is no active trace (previously always returned true even for INVALID spans).
- Fix `Traces.trace_context` to return nil when there is no active trace (previously returned invalid Context objects with zero trace_ids).
- Both methods now use `OpenTelemetry::Trace::Span::INVALID` for proper detection of inactive traces instead of manual trace_id checking.
- Add test case to ensure `trace_context` returns nil in clean Fiber context.
- Resolves issues with Async incorrectly detecting active traces.

## Types of Changes

- Bug fix.

## Contribution

- [x] I added tests for my changes.
- [x] I tested my changes locally.
- [x] I agree to the [Developer's Certificate of Origin 1.1](https://developercertificate.org/).

### CODE REVIEW

File: gems.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@
 	gem "sus"
 	gem "covered"
 	gem "decode"
-
+
```
Comment (simi) [inline gems.rb:24]: :scissors:
